### PR TITLE
[zh-cn] Update 'working with the history api'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

reflect history API rewrite changes from upstream (https://github.com/mdn/content/pull/30354)

### CODE REVIEW

File: files/zh-cn/web/api/history_api/working_with_the_history_api/index.md [hunk 1]
```diff
@@ -1,105 +1,192 @@
...
-  - : 新的历史条目的 URL 是由这个参数给出的。注意，在调用 `pushState()` 后，浏览器不会尝试加载这个 URL，但它可能会在以后尝试加载这个 URL，例如在用户重新启动浏览器后。新的 URL 不需要是绝对的；如果它是相对的，它就会相对于当前的 URL 进行解析。新的 URL 必须与当前的 URL 同源；否则，`pushState()` 将抛出一个异常。这个参数是可选的；如果没有指定它，它将被设置为文档的当前 URL。
+- 如果只需要更新页面的一部分，每次加载整个页面的效率会很低。
+- 跨页面导航时很难保持应用程序状态
 
-在某种意义上，调用 `pushState()` 类似于设置 `window.location = "#foo"`，因为两者都将创建并激活与当前文档相关的另一个历史条目。
+由于这些原因，{{Glossary("SPA", "单页应用")}}（SPA）是一种流行的 web 应用程序模式，在这种模式中，网站由一个页面组成，当用户点击链接时，页面会：
 
-但是 `pushState()` 有几条优势：
+1. 阻止加载新页面的默认行为
+2. {{domxref("fetch()", "获取", "", "nocode")}}显示新内容
```
Comment (yin1999) [inline files/zh-cn/web/api/history_api/working_with_the_history_api/index.md:43]:
```suggestion
2. {{domxref("fetch()", "获取", "", "nocode")}}要显示的新内容
```

File: files/zh-cn/web/api/history_api/working_with_the_history_api/index.md [hunk 2]
```diff
@@ -1,105 +1,192 @@
...
+      displayContent(json);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+});
+```
 
-在其他文档中，它创建了一个具有 `null` 命名空间 URI 的元素。
+在该点击处理程序中，如果链接包含数据属性 `"data-creature"`，我们就会使用该属性的值来获取包含页面新内容的 JSON 文件。
```
Comment (yin1999) [inline files/zh-cn/web/api/history_api/working_with_the_history_api/index.md:67]:
```suggestion
在该点击处理器中，如果链接包含数据属性 `"data-creature"`，我们就会使用该属性的值来获取包含页面新内容的 JSON 文件。
```

File: files/zh-cn/web/api/history_api/working_with_the_history_api/index.md [hunk 3]
```diff
@@ -1,105 +1,192 @@
...
 ```
 
-上面这两行的解释可以在上面的 [_pushState() 方法的例子_](#pushstate_方法示例)部分找到。
+问题在于，它破坏了浏览器“后退”和“前进”按钮的预期行为。
+
+从用户的角度来看，他们点击了一个链接，页面就更新了，所以看起来像是一个新页面。如果用户按下浏览器的“后退”按钮，就会返回到点击链接之前的状态。
+
+但在浏览器看来，最后一个链接并没有加载新页面，因此“后退”按钮会将浏览器带回到用户打开 SPA 之前加载的页面。
+
+这正是 `pushState()`、`replaceState()` 和 `popstate` 事件所要解决的问题。它们使我们能够合成历史条目，并在当前会话历史条目更改为这些条目之一（例如，由于用户按下了“后退”或“前进”按钮）时获得通知。
```
Comment (yin1999) [inline files/zh-cn/web/api/history_api/working_with_the_history_api/index.md:104]:
```suggestion
这正是 `pushState()`、`replaceState()` 和 `popstate` 事件所要解决的问题。它们使我们能够合成历史条目，并在当前会话历史条目更改为这些条目之一（例如，由于用户按下了“后退”或“前进”按钮）时收到通知。
```

File: files/zh-cn/web/api/history_api/working_with_the_history_api/index.md [hunk 4]
```diff
@@ -1,105 +1,192 @@
...
+  }
+});
 ```
 
-这将导致 URL 栏显示 `https://mozilla.org/bar2.html`，但不会导致浏览器加载 `bar2.html` 或甚至检查 `bar2.html` 是否存在。
+这里，我们使用三个参数调用了 `pushState()`：
+
+- `json`：这是我们刚刚获取的内容。它将与历史条目一起存储，之后将作为 {{domxref("PopStateEvent.state", "state")}} 属性包含在传递给 `popstate` 事件处理器的参数中。
+- `""`：这是与传统网站向后兼容所必需的，应始终为空字符串。
+- `creature`：这将用作条目的 URL。它将显示在浏览器的 URL 栏中，并在页面发出的任何 HTTP 请求中用作 {{httpheader("Referer")}} 标头的值。请注意，这必须与页面{{Glossary("Same-origin policy", "同源")}}} 。
```
Comment (yin1999) [inline files/zh-cn/web/api/history_api/working_with_the_history_api/index.md:133]:
```suggestion
- `creature`：这将用作条目的 URL。它将显示在浏览器的 URL 栏中，并在页面发出的任何 HTTP 请求中用作 {{httpheader("Referer")}} 标头的值。请注意，这必须与页面{{Glossary("Same-origin policy", "同源")}} 。
```

File: files/zh-cn/web/api/history_api/working_with_the_history_api/index.md [hunk 5]
```diff
@@ -1,105 +1,192 @@
...
 
-`history.replaceState()` 的操作与 `history.pushState()` 完全一样，只是 `replaceState()` 修改了当前的历史条目，而不是创建一个新的。注意，这并不妨碍在全局浏览器历史中创建一个新条目。
+```json
+{
+  "description": "秃鹫并不“秃”。",
+  "image": {
+    "src": "images/eagle.jpg",
+    "alt": "一只秃鹫"
+  },
+  "name": "秃鹫"
```
Comment (yin1999) [inline files/zh-cn/web/api/history_api/working_with_the_history_api/index.md:78]: Let's use the untranslated content (or this example is inconsistent with the live example)

**Review by yin1999 — COMMENTED** [review]
Great. This page once introduced me to the history API, now it's time to update it.

Let's fix some nits before merging :)

File: files/zh-cn/web/api/history_api/working_with_the_history_api/index.md (hunk 5 above)
Comment (jasonren0403) [inline files/zh-cn/web/api/history_api/working_with_the_history_api/index.md:78]:
```suggestion
  "description": "Bald eagles are not actually bald.",
  "image": {
    "src": "images/eagle.jpg",
    "alt": "A bald eagle"
  },
  "name": "Eagle"
```

**Review by yin1999 — APPROVED** [review]
Thank you @jasonren0403

**Pull request merged**